src/witan/httpapi/spec.clj

```diff
@@ -31,128 +39,75 @@
-        (time-parser x)
-        ::s/invalid)
-      (catch IllegalArgumentException e
-        ::s/invalid))))
-
-(def timestamp? (s/conformer -timestamp? identity))
-
-(defn -varint?
-  [x]
-  (cond
-    (int? x) x
-    (instance? clojure.lang.BigInt x) x
-    :else ::s/invalid))
-
-(def varint?
-  "big or small int"
-  (s/conformer -varint? identity))
-
-(defn email?
-  [s]
-  (when-not (clojure.string/blank? s)
-    (re-find #"^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,63}" s)))
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Specs
 
-(s/def ::id (api-spec uuid? "string"))
-
-;; Auth
-(s/def ::password spec/string?)
-(s/def ::username (api-spec email? "string"))
-(s/def ::auth-token spec/string?)
-(s/def ::refresh-token spec/string?)
-(s/def ::token-pair (s/keys :req-un [::auth-token ::refresh-token]))
-(s/def ::token-pair-container (s/keys :req-un [::token-pair]))
-
-;; User
-(s/def :kixi.user/id (api-spec uuid? "string"))
-
-;; Permissions
-(s/def :kixi.datastore.metadatastore/meta-read (s/coll-of spec/string?))
-(s/def :kixi.datastore.metadatastore/meta-update (s/coll-of spec/string?))
-(s/def :kixi.datastore.metadatastore/file-read (s/coll-of spec/string?))
-
-;; Metadata
-(s/def :kixi.datastore.metadatastore/size-bytes (api-spec varint? "integer"))
-(s/def :kixi.datastore.metadatastore/file-type spec/string?)
-(s/def :kixi.datastore.metadatastore/header spec/boolean?)
-(s/def :kixi.datastore.metadatastore/name spec/string?)
-(s/def :kixi.datastore.metadatastore/id (api-spec uuid? "string"))
-(s/def :kixi.datastore.metadatastore/type spec/string?)
-(s/def :kixi.datastore.metadatastore/description spec/string?)
-(s/def :kixi.datastore.metadatastore/source spec/string?)
-(s/def :kixi.datastore.metadatastore/created (api-spec timestamp? "string"))
-
-(s/def :kixi.datastore.metadatastore/provenance (s/keys :req [:kixi.datastore.metadatastore/source
-                                                              :kixi.datastore.metadatastore/created
-                                                              :kixi.user/id]))
-(s/def :kixi.datastore.metadatastore/sharing (s/keys :opt [:kixi.datastore.metadatastore/meta-read
-                                                           :kixi.datastore.metadatastore/meta-update
-                                                           :kixi.datastore.metadatastore/file-read]))
-(s/def :kixi.datastore.filestore/id (api-spec uuid? "string"))
-
-(s/def ::file-metadata
-  (s/keys :req [:kixi.datastore.metadatastore/size-bytes
-                :kixi.datastore.metadatastore/file-type
-                :kixi.datastore.metadatastore/header
-                :kixi.datastore.metadatastore/name
-                :kixi.datastore.metadatastore/id
-                :kixi.datastore.metadatastore/type
-                :kixi.datastore.metadatastore/description
-                :kixi.datastore.metadatastore/provenance]))
+(s/def ::id (api-spec sc/uuid? "string"))
```

Comment:
I take it this is all internal spec and doesn't need to live in the shared spec library?

Reply:
Correct